.github/actions/archive-charm-testing-artifacts/action.yaml

```diff
@@ -0,0 +1,33 @@
+name: archive-charm-testing-artifacts
+description: Archives the charm(s) that were built during a job, along with possible log and juju crashdump outputs
+inputs:
+  enable_containerd_template_adjustment:
+    description: 'Whether to adjust containerd template for GHCR.io credentials'
+    required: false
+  ghcr_package_api_user:
+    description: 'GitHub Container Registry API User'
+    required: false
+  ghcr_package_api_token:
+    description: 'GitHub Container Registry API Token'
+    required: false
+    
+runs:
+  using: 'composite'
+  steps:
+  - name: Archive the tested charm
+    uses: actions/upload-artifact@v3
+    with:
+      name: tested-charm
+      path: .tox/**/*.charm
+      retention-days: 5
+  - name: Archive logs on failure
+    uses: actions/upload-artifact@v3
+    with:
+      name: charmcraft-logs
+      path: ./frontend/charm/**/*.log
+  - name: Archive juju crashdump on failure
+    if: failure()
+    uses: actions/upload-artifact@v3
+    with:
+      name: juju-crashdump
+      path: ./frontend/charm/juju-crashdump-*.tar.xz
```

Comment:
the path `./frontend/charm` I assume relates to test observer's frontend charm. But since this action is in a separate repo it should be generic and works with any workflow right?

Reply:
Oops, fixed too.